src/map/npc.c

```diff
@@ -3256,7 +3267,20 @@ int npc_instanceinit(struct npc_data* nd)
 	struct event_data *ev;
 	char evname[EVENT_NAME_LENGTH];
 
-	snprintf(evname, ARRAYLENGTH(evname), "%s::OnInstanceInit", nd->exname);
+	snprintf(evname, ARRAYLENGTH(evname), "%s::%s", nd->exname, script_config.instance_init_event_name);
+
+	if( ( ev = (struct event_data*)strdb_get(ev_db, evname) ) )
+		run_script(nd->u.scr.script,ev->pos,0,nd->bl.id);
+
+	return 0;
+}
+
+int npc_instancedestroy(struct npc_data* nd)
+{
+	struct event_data *ev;
+	char evname[EVENT_NAME_LENGTH];
+
```

Comment:
I wonder if a nullpo on nd wouldn't be safer here. Or it may be totally useless..